pages/index.vue

```diff
@@ -1,32 +1,90 @@
 <script setup lang="ts">
-interface markerData {
-  latitude: number;
-  longitude: number;
-  name: string;
-  description: string;
-  rating: number;
+import type { MapLocation } from '~/types/mapLocation';
+import type { Coordinate } from '~/types/coordinate';
+
+const mapId = 'mapId';
+const savedLocations = ref<Array<MapLocation>>([]);
+const lastClickedCoordinate = ref<Coordinate | null>(null);
+
+function saveLocation() {
+  if (!lastClickedCoordinate.value) return;
+
+  savedLocations.value.push({
+    coordinates: {
+      latitude: lastClickedCoordinate.value.latitude,
+      longitude: lastClickedCoordinate.value.longitude,
+    },
+    name: 'ooh la la',
+    description: 'nice',
+    rating: 69,
+  });
+
+  lastClickedCoordinate.value = null;
 }
 
-const savedMarkers = ref([]);
+function deleteLocation(location: MapLocation) {
+  return (savedLocations.value = savedLocations.value.filter(
+    l => l.coordinates.latitude !== location.coordinates.latitude
```

Comment:
Should this also check the longitude?

Reply:
similar reason for why the key is just latitude, the value is v precise so it's very unlikely to have two latitudes the same. Will make it a `lat` and `long` check tho